core/src/main/scala/org/apache/spark/TaskContext.scala

```diff
@@ -183,6 +183,15 @@ abstract class TaskContext extends Serializable {
   @Evolving
   def resources(): Map[String, ResourceInformation]
 
+  /**
+   * The shuffle generation ID of the stage that this task belongs to, it returns the stage
+   * attempt number while the stage is not determinate and returns -1 on the contrary.
+   */
+  def getShuffleGenerationId(shuffleId: Int): Int = {
```

Comment:
`TaskContext` is public, this method should be `private[spark]`

Reply:
Thanks, done in 4c4af33.